client-v2/src/actions/Collections.ts

```diff
@@ -464,37 +464,47 @@ function publishCollection(
         );
 
         dispatch(getCollections([collectionId]));
+        dispatch(pollForCollectionPublished(collectionId, frontId));
```

Comment:
Extracting this side effect code and pulling into a new function to clarify a) what it does and b) that the main function doesn't have to wait for the side effect to complete before it returns